skins/cat17/address-form/src/components/Name.vue

```diff
@@ -0,0 +1,44 @@
+<template>
+<div class="row">
+    <div v-if="isCompany" class="field-grp field-company col-xs-12 col-md-6">
+        <label for="company-name">{{ messages.companyname_label }}</label>
+        <input type="text" 
+                id="company-name" 
+                name="companyName" 
+                :placeholder="messages.companyname_label" 
+                data-pattern="^.+$" 
+                @blur="validateInput('companyName')">
+        <span v-if="error('companyName')" class="error-text">{{ messages.form_companyname_error }}</span>
+    </div>
+    <div v-else>
+        <div class="field-grp field-firstname col-xs-12 col-md-6">
+            <label for="first-name">{{ messages.firstname_label }}</label>
+            <input type="text" 
+                    id="first-name" 
+                    name="firstName" 
+                    :placeholder="messages.firstname_label" 
+                    data-pattern="^.+$"
+                    @blur="validateInput('firstName')">
```

Comment:
the `data-pattern` was an attribute that catered more to the jQuery-based code. With Vue and Vuex you can use the validation pattern as a parameter to `validateInput`.